src/pages/index.js

```diff
@@ -92,20 +114,73 @@ import { PopupWithImage } from "../components/PopupWithImage";
     `#${addForm.parentElement.id}`,
     handleProfileAddSubmit
   );
-  const userInfo = new UserInfo(profileName, profileDescription);
 
   imagePopup.setEventListeners();
 
+  //preset profile info on load
+  api.getProfileInfo().then((results) => {
+    console.log(results);
+    profileName.textContent = results.name;
+    profileDescription.textContent = results.about;
+    avatarPhoto.src = results.avatar;
+  });
+
   //functions
+  const deletePopup = new confirmationPopup("#confirmation-modal");
+  deletePopup.setEventListeners();
+
+  function addCard(name, description, template, id) {
+
+    const modelCard = new Card(
+      name,
+      description,
+      template,
+      imageClickHandler,
+      (card) => {
+        console.log(card._id);
+
+        // api.deleteCard(card._id, card.deleteCard()); // 
+        deletePopup.open();
+        deletePopup.setSubmitAction(() =>{
+          api.deleteCard(card._id).then(() =>{
+            card.deleteCard();
+            deletePopup.close();
+          })
+        })
+
+      },
+      toggleLikeButton,
+      initialCardsInfo,
+      id
+    );
+
+    galleryList.prepend(modelCard.addCard());
+  }
 
   function createCard(name, description, template) {
-    const modelCard = new Card(name, description, template, imageClickHandler);
-    cardsSection.addItem(modelCard.addCard());
+    api.addCardInfo(name, description, (name, link, id) => {
+      console.log(template);
+      addCard(name, link, template, id);
+    }).then(
+      addPopup.saveButton.textContent = "Saving..."
```

Comment:
In blocks `then`, `catch`, `finally` you need to pass only the **reference** to the function (not a function call) otherwise the function is called immediately without waiting for the response from the server. You can pass an **unnamed** arrow function as well. 

And you need to start the loading effect **before** the request (not in `then`)


**Please, correct it everywhere**